modules/configuration/src/main/java/org/apache/ignite/internal/configuration/util/ConfigurationUtil.java

```diff
@@ -1127,6 +1128,12 @@ public static boolean containsNameAnnotation(Field schemaField) {
 
         assert property instanceof ConfigurationNode;
 
+        // If the property is set as {@code true} the attempt to get configuration property direct from Metastorage will be skipped, and
+        // the local one local will be returned. Otherwise, when the property is set as {@code false}, the direct property will be returned.
+        if (IgniteSystemProperties.getBoolean("IGNITE_GET_METADATA_LOCALLY_ONLY")) {
```

Comment:
We should not read this system property through `IgniteSystemProperties` because this call ends up with `Hashtable.get()` which is `synchronized`.